packages/next/src/server/app-render/create-error-handler.tsx

```diff
@@ -204,10 +207,20 @@ export function createHTMLErrorHandler(
       }
 
       if (!silenceLogger) {
-        onHTMLRenderError(err)
+        if (isRSCError) {
+          onHTMLRenderRSCError(err)
+        } else {
+          onHTMLRenderSSRError(err)
+        }
```

Comment:
it may be that this is how it has to be. but what I was suggesting earlier was that you move `onHTMLRenderRSCError` into the `createHTMLReactServerErrorHandler` instead of leaving it in `createHTMLErrorHandler`. This way you don't need a boolean flag like that and you simply report RSC errors in the RSC handler and SSR errors in the SSR handler. Is that not possible for some reason?